web/templates/js/dingEventsSlide.js

```diff
@@ -155,31 +155,34 @@ if (!window.slideFunctions['ding-events']) {
     run: function runDingEventsSlide(slide, region) {
       region.itkLog.info("Running ding events slide: " + slide.title);
 
-      var slide_duration = slide.options.rss_duration ? slide.options.rss_duration : 15;
-
-      // Check that external_data exists, if not stay on for one duration and
-      // continue to next slide.
-      if (!slide.external_data || !slide.external_data.events || slide.external_data.events.length <= 0) {
-        // Go straight to the next slide if we don't have any data. For now this
-        // simply assumes that we have a "next" to go to, if not, we're going
+      // Experience has shown that we can't be certain that all our data is
+      // present, so we'll have to be careful verify presence before accessing
+      // anything.
+      if (!slide.options || !slide.external_data || !slide.external_data.events || slide.external_data.events.length <= 0 || !slide.event_settings) {
```

Comment:
Why do we need the check for `slide.options` and `slide.options` here? Neither seems to be referenced within the scope of this `if` statement ending on L180.

However `slide.options` *is* referenced later on on L184.

Reply:
The theory is: If we're missing slide.options something is wrong and we should bail outwhich is why we check for !slide.options and return if it resolves as true.

Reply:
Ah - right. This is the other way around compared to how I immediately perceived it. This is checks for missing data and immediate ´handling of that situation - not a guard against missing data before executing expected business logic.